scripts/policy/protocols/conn/known-services.zeek

```diff
@@ -89,12 +111,16 @@ event zeek_init()
 	}
 
 event service_info_commit(info: ServicesInfo)
-                          
 	{
 	if ( ! Known::use_service_store )
 		return;
 
-	local key = AddrPortPair($host = info$host, $p = info$port_num);
+	local v : vector of string;
+	for ( s in info$service )
+		v += s;
+	sort(v, strcmp);	# sort the vector for proper key comparison in put_unique
+
+	local key = AddrPortServTriplet($host = info$host, $p = info$port_num, $serv = v);
 
 	when ( local r = Broker::put_unique(Known::service_store$store, key,
 	                                    T, Known::service_store_expiry) )
```

Comment:
This data store seems only used for log de-duplication purposes at the moment, so you might just do a loop over each service and try `put_unique` for each key that is of the form `(host, port, single-service)`.  Changing it to that would then make the log output consistent with the other implementation of known-services that only uses partitioning/events -- e.g. both would log individual entries for either HTTP and SSL instead of having the data store version log just HTTP then also HTTP,SSL.